internal/dataplane/parser/translate_httproute_test.go

```diff
@@ -1352,6 +1352,96 @@ func TestGetHTTPRouteHostnamesAsSliceOfStringPointers(t *testing.T) {
 	}
 }
 
+func TestGeneratePluginsFromHTTPRouteFilters(t *testing.T) {
+	testCases := []struct {
+		name            string
+		filters         []gatewayv1beta1.HTTPRouteFilter
+		path            string
+		expectedPlugins []kong.Plugin
+	}{
+		{
+			name:            "no filters",
+			filters:         []gatewayv1beta1.HTTPRouteFilter{},
+			expectedPlugins: []kong.Plugin{},
+		},
+		{
+			name: "request header modifier",
+			filters: []gatewayv1beta1.HTTPRouteFilter{
+				{
+					Type: gatewayv1beta1.HTTPRouteFilterRequestHeaderModifier,
+					RequestHeaderModifier: &gatewayv1beta1.HTTPHeaderFilter{
+						Set: []gatewayv1beta1.HTTPHeader{
+							{
+								Name:  "header-to-set",
+								Value: "bar",
+							},
+						},
+						Remove: []string{"header-to-remove"},
+					},
+				},
+			},
+			expectedPlugins: []kong.Plugin{
+				{
+					Name: kong.String("request-transformer"),
+					Config: kong.Configuration{
+						"add": map[string][]string{
+							"headers": {
+								"header-to-set:bar",
+							},
+						},
+						"remove": map[string][]string{
+							"headers": {
+								"header-to-remove",
+							},
+						},
+						"replace": map[string][]string{
+							"headers": {
+								"header-to-set:bar",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "request redirect modifier",
+			filters: []gatewayv1beta1.HTTPRouteFilter{
+				{
+					Type: gatewayv1beta1.HTTPRouteFilterRequestRedirect,
+					RequestRedirect: &gatewayv1beta1.HTTPRequestRedirectFilter{
+						Hostname:   (*gatewayv1beta1.PreciseHostname)(lo.ToPtr("example.org")),
+						StatusCode: lo.ToPtr(302),
+					},
+				},
+			},
+			path: "/test",
+			expectedPlugins: []kong.Plugin{
+				{
+					Name: kong.String("request-termination"),
+					Config: kong.Configuration{
+						"status_code": lo.ToPtr(302),
+					},
+				},
+				{
+					Name: kong.String("response-transformer"),
+					Config: kong.Configuration{
+						"add": map[string][]string{
+							"headers": {
+								"Location: http://example.org:80/test",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		plugins := generatePluginsFromHTTPRouteFilters(tc.filters, tc.path, nil)
+		assert.Equal(t, tc.expectedPlugins, plugins, tc.name)
```

Comment:
Nit: from readability standpoint, better to use `t.Run(tc.name, ... )` than running all tests cases in 1 test. This way we'll get separation of tests and see whether each test passes or failed (not only failures). WDYT?